src/enochecker/enochecker.py

```diff
@@ -359,34 +218,42 @@ def _setup_logger(self) -> None:
         self.critical: Callable[..., None] = self.logger.critical
 
     @property
-    def current_round(self) -> Optional[int]:
+    def ctx(self) -> str:
         """
-        Deprecated! Only for backwards compatibility! Use self.round instead.
+        short hand version for self.task_chain_id
 
-        :return: current round
+        :return: self.task_chain_id
         """
-        warnings.warn(
-            "current_round is deprecated, use round instead", DeprecationWarning
-        )
-        return self.round
+        return self.task_chain_id
 
     @property
-    def noise(self) -> Optional[str]:
+    def noise(self) -> str:
         """
-        Pretty similar to a flag, just in a different mode (storeNoise vs storeFlag).
+        Creates a stable noise value for the current context.
+        Do not use for indexing (use self.ctx instead).
 
-        :return: The noise
+        :return: A noise string, unique for each ctx.
         """
-        return self.flag
+        if self._noise_cache is None:
+            if not self.ctx:
+                self.warning("No valid ctx when calling noise!")
+                return "<none>"
+            # We cache the hex in case it's called often.
+            m = hashlib.sha256()
+            m.update(self.ctx.encode())
+            self._noise_cache = m.hexdigest()
+        return self._noise_cache
 
     @property
-    def time_running(self) -> float:
+    def time_running(self) -> int:
         """
         How long this checker has been running for.
 
-        :return: time this checker has been running for
+        :return: time this checker has been running for in milliseconds
```

Comment:
Don't we want second floats inside python at least? Most APIs you encounter use seconds.

Reply:
It seems there are a lot of inconsistencies as far as timeouts are concerned throughout the lib, SimpleSocket expects an int timeout instead of a float but that is still seconds

Reply:
done. although there are a bunch of `# type: ignore` we can get rid of once this change eventually ends up in a mypy release: https://github.com/python/typeshed/pull/5243